### ISSUE TITLE
GUI error: Unrecognized property 'Units' for class 'matlab.ui.controls.ToolbarStateButton'

### ISSUE THREAD

**Issue by Kevin-Mattheus-Moerman** [hooman650/BioSigKit#3]
I am running MATLAB 2018b on Linux (Ubuntu 18.04).

When I run `RunBioSigKit` and use the import button to load the example data `ECG1.mat` I get the following error messages:

```
Warning: Error occurred while executing the listener callback for event ChildAdded defined for class uix.ChildObserver:
Unrecognized property 'Units' for class 'matlab.ui.controls.ToolbarStateButton'.

Error in uix.setPosition (line 13)
o.Units = u;

Error in uix.Panel/redraw (line 56)
                uix.setPosition( obj.Contents_(selection), position, 'pixels' )

Error in uix.mixin.Container/set.Dirty (line 132)
                    obj.redraw() % redraw now

Error in uix.mixin.Container/addChild (line 222)
            obj.Dirty = true;

Error in uix.mixin.Panel/addChild (line 88)
            addChild@uix.mixin.Container( obj, child )

Error in uix.mixin.Container/onChildAdded (line 162)
            obj.addChild( eventData.Child )

Error in uix.mixin.Container>@(varargin)obj.onChildAdded(varargin{:}) (line 51)
                childObserver, 'ChildAdded', @obj.onChildAdded );

Error in uix.ChildObserver/addChild (line 109)
                notify( obj, 'ChildAdded', uix.ChildEvent( oChild ) )

Error in uix.ChildObserver/addChild (line 116)
                    obj.addChild( nChild, oGrandchildren(ii) )

Error in uix.ChildObserver/addChild (line 116)
                    obj.addChild( nChild, oGrandchildren(ii) )

Error in uix.ChildObserver/addChild (line 116)
                    obj.addChild( nChild, oGrandchildren(ii) )

Error in uix.ChildObserver/addChild (line 116)
                    obj.addChild( nChild, oGrandchildren(ii) )

Error in uix.ChildObserver/addChild (line 116)
                    obj.addChild( nChild, oGrandchildren(ii) )

Error in uix.ChildObserver/addChild (line 116)
                    obj.addChild( nChild, oGrandchildren(ii) )

Error in uix.ChildObserver/addChild (line 116)
                    obj.addChild( nChild, oGrandchildren(ii) )

Error in uix.ChildObserver/addChild (line 116)
                    obj.addChild( nChild, oGrandchildren(ii) )

Error in uix.ChildObserver>@(~,e)obj.addChild(nChild,e.Child) (line 96)
                    @(~,e)obj.addChild(nChild,e.Child) );

Error in matlab.graphics.controls.AxesToolbar/parentToolbarToAxesPane

Error in matlab.graphics.controls.ToolbarController/handleMouseMotion

Error in matlab.graphics.controls.ToolbarController>@(e,d)obj.handleMouseMotion(e,d) 
> In uix.ChildObserver/addChild (line 109)
  In uix.ChildObserver/addChild (line 116)
  In uix.ChildObserver/addChild (line 116)
  In uix.ChildObserver/addChild (line 116)
  In uix.ChildObserver/addChild (line 116)
  In uix.ChildObserver/addChild (line 116)
  In uix.ChildObserver/addChild (line 116)
  In uix.ChildObserver/addChild (line 116)
  In uix.ChildObserver/addChild (line 116)
  In uix.ChildObserver>@(~,e)obj.addChild(nChild,e.Child) (line 96)
  In matlab.graphics.controls.AxesToolbar/parentToolbarToAxesPane
  In matlab.graphics.controls.ToolbarController/handleMouseMotion
  In matlab.graphics.controls.ToolbarController>@(e,d)obj.handleMouseMotion(e,d) 
```

**Comment by hooman650** [comment]
Hi @Kevin-Mattheus-Moerman , I have been trying to trace back this issue, unfortunately I am on a Windows OS machine and only use ubuntu terminal which doesnt support visualization... From a first glance it seems like a third party error (I use [GUI toolbox libraries](https://www.mathworks.com/matlabcentral/fileexchange/47982-gui-layout-toolbox) for the GUI part). But I will keep working on it and update you soon.

**Comment by hooman650** [comment]
In fact, this issue seems to be a third party issue ([GUI Layout Matlab](https://www.mathworks.com/matlabcentral/fileexchange/47982-gui-layout-toolbox)). I have updated the affected library with its latest version hoping that the bug might be fixed. Could you please check again and update me? If not I will post the bug on their forum and try to figure it out... Thanks

**Comment by Kevin-Mattheus-Moerman** [comment]
@hooman650 the bug remains I'm afraid.

The source of the error seems to be the `setPosition` function of that 3rd party package you mentioned. I only get this error if no data is loaded and no analysis is performed yet and when I move the mouse over the graphing area for the signal or the display area for the stats. So perhaps when the signals to be plotted and the stats to be displayed are not there/empty then there is no `units` property defined on the input for the `setPosition` function. In particular I get
```
Unrecognized property 'Units' for class 'matlab.ui.controls.ToolbarStateButton'.
Error in uix.setPosition (line 13)
o.Units = u;
```
If I look at the `setPosition` function and inspect, using `fieldnames(o)` what properties the variable `o` has when the error occurred I found: 
```
    {'ValueChangedFcn' }
    {'Value'           }
    {'Tooltip'         }
    {'Icon'            }
    {'Children'        }
    {'Parent'          }
    {'Visible'         }
    {'HandleVisibility'}
    {'UIContextMenu'   }
    {'ButtonDownFcn'   }
    {'BusyAction'      }
    {'BeingDeleted'    }
    {'Interruptible'   }
    {'CreateFcn'       }
    {'DeleteFcn'       }
    {'Type'            }
    {'Tag'             }
    {'UserData'        }
```
So at this point, when the signal/stats areas are empty, there are no `units` or `position` properties for the object being treated. Perhaps this case can easily be detected to avoid the error. 

I've just tested this on a Windows machine with MATLAB 2018a and the error did not occur. It does still occur on my Ubuntu 18.04 system with MATLAB 2018b. So I wonder if this is MATLAB version issue. 

Since this error only seems to occur for my system I'll not make this issue a requirement for JOSS publication.
